modules/service/src/main/scala/lucuma/odb/api/service/Routes.scala

```diff
@@ -100,12 +103,45 @@ object Routes {
         resp   <- toResponse(result)
       } yield resp
 
-    val keepAliveStream: Stream[F, FromServer] =
-      Stream
-        .constant[F, FromServer](FromServer.ConnectionKeepAlive)
-        .metered(KeepAliveDuration)
+    val webSocketConnection: F[Response[F]] = {
+
+      val keepAliveStream: Stream[F, FromServer] =
+        Stream
+          .constant[F, FromServer](FromServer.ConnectionKeepAlive)
+          .metered(KeepAliveDuration)
+
+      def logFromServer(user: F[Option[User]], msg: FromServer): F[Unit] =
+        user.flatMap { u =>
+          msg match {
+            case FromServer.ConnectionKeepAlive =>
+              debug(u, s"Sending ConnectionKeepAlive")
+            case _                              =>
+              info(u, s"Sending to client: ${trimmedMessage(msg)}")
+          }
+        }
+
+      def logTextFrame(user: F[Option[User]], f: WebSocketFrame): F[Unit] = {
+
+        // The connection_init message payload has authorization information
+        // which should not be logged.
+        val AuthRegEx    = """"Authorization":\s*"[^"]*"""".r.unanchored
+        val RedactedAuth = """"Authorization": <REDACTED>"""
```

Comment:
This is a super tiny detail, but if you don't want to repeat `Authorization` I think you can capture it in `AuthRegEx` (`("Authorization":)`) and then use `$1` in `RedactedAuth`.

Reply:
Thanks, that was a good idea.  :+1: